### PR TITLE
Bluetooth: rename Kconfig.iso

### PR DESCRIPTION
**Pull request by quic-egmc**
Some git systems employ hooks to prevent binary files from being committed accidentally.  The Kconfig.iso file, which contains Kconfig flags related to isochronous operation, causes the file-extension based binary detection hooks to falsely detect this is a binary file and reject pushes.

Rename Kconfig.iso toKconfig.isoc to avoid this issue in forks of zephyr.

### CODE REVIEW

**Comment by jhedberg** [comment]
I suppose this is fine. The one concern I have is that this is not the conventional short form, and e.g. all the Kconfig symbols in that file have a `BT_ISO_*` prefix. I fear that someone will at some point in the future forget about the reasoning for this change and rename the file back to `Kconfig.iso` in the name of consistency.

**Comment by quic-egmc** [comment]
> I suppose this is fine. The one concern I have is that this is not the conventional short form, and e.g. all the Kconfig symbols in that file have a `BT_ISO_*` prefix. I fear that someone will at some point in the future forget about the reasoning for this change and rename the file back to `Kconfig.iso` in the name of consistency.

Fair point - I took the easy path as this was only a two file change but changing the config flags would be more impactful.  Let me know if you'd like me to change the extension and/or config flags to align.

**Comment by Thalley** [comment]
@quic-egmc Can you elaborate on what hooks may have an issue with `.iso`? 

The naming convention of `Kconfig.<feature>` is something that we use quite a lot in LE Audio for example (see the Kconfig files in https://github.com/zephyrproject-rtos/zephyr/tree/main/subsys/bluetooth/audio), and if this is a generic issue, then we may want to simply rename the Kconfig files to e.g. `Kconfig_<feature>` instead.

**Comment by MaureenHelm** [comment]
@quic-egmc do you plan to return to this PR?

**Comment by quic-egmc** [comment]
Sorry for neglecting the PR...

> Can you elaborate on what hooks may have an issue with .iso?

This was a tool that was being "helpful" by trying to prevent binaries from being added to git repos.  AFAIK only this file was problematic in this regard.

**Comment by Thalley** [comment]
> @Thalley (fyi @MaureenHelm ) Sorry for neglecting the PR...
> 
> > Can you elaborate on what hooks may have an issue with .iso?
> 
> This was a tool that was being "helpful" by trying to prevent binaries from being added to git repos. AFAIK only this file was problematic in this regard.

I see, that makes sense given the history of the `.iso` file extension. 

I don't think we are planning to rename to `isoc` as that is not a common abbreviation for isochronous channels. 

If we do decide to rename it to e.g. `Kconfig_iso` instead to avoid such file name extension issues, then we should do that across all of course Kconfig files, which would require additional input from others. 

@alwa-nordic @jhedberg as maintainers of the BT subsystem, what are your thoughts on this? Any suggestions for solutions?

**Comment by alwa-nordic** [comment]
> If we do decide to rename it to e.g. `Kconfig_iso` instead to avoid such file name extension issues, then we should do that across all of course Kconfig files, which would require additional input from others.
> 
> @alwa-nordic @jhedberg as maintainers of the BT subsystem, what are your thoughts on this? Any suggestions for solutions?

I agree with you. This is an issue for the Kconfig subsystem maintainers to address. I.e. The issue to be addressed is that users of Zephyr may have tools that confuse filename extensions with Kconfig-files having a 'Kconfig.' prefix and no filename extension.

I think we should not break pattern with the rest of the project. I suspect that we have tools that look for the pattern 'Kconfig.' in filenames.

We cannot on a general basis accommodate everyone's downstream linters. (Especially regarding false positives.) But I encourage the Kconfig maintainers to consider that filename extensions are a useful de-facto standard and look at the possibility of giving Kconfig files an extension.

I could go for renaming our Kconfigs to have fuller paths '**Kconfig.bt_iso**'. That's anyway nice and fixes the problem in this particular case.

**Comment by Thalley** [comment]
> I could go for renaming our Kconfigs to have fuller paths '**Kconfig.bt_iso**'. That's anyway nice and fixes the problem in this particular case.

That is a good idea. The `bt_` extension prefix will be a nice generic solution, and can even be applied to https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/Kconfig to be `Kconfig.bt`

If we decide to rename, I'd be in favor of this approach.

**Comment by quic-egmc** [comment]
> > I could go for renaming our Kconfigs to have fuller paths '**Kconfig.bt_iso**'. That's anyway nice and fixes the problem in this particular case.
> 
> That is a good idea. The `bt_` extension prefix will be a nice generic solution, and can even be applied to https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/Kconfig to be `Kconfig.bt`
> 
> If we decide to rename, I'd be in favor of this approach.
> 
> @quic-egmc Can you verify that `Kconfig.bt_iso` would not be an issue for your system?

Somehow I missed this thread long ago - no issue here.

**Comment by Thalley** [comment]
@alwa-nordic @jori-nordic Should we go ahead and add the `bt_` prefix to the extension of Kconfig files in BT? Maybe omit it for https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/Kconfig since it's extensionless

**Comment by Thalley** [comment]
> Replacing the `.` with a `_` should also work fine, and might be a better alternative, since `.` usually indicates a file extension, which is not the case here.

This might actually be the better approach, given that `.` usually indicates a file extension, which isn't the case here.

**Comment by Thalley** [comment]
@quic-egmc I'd be ok with approving this PR if you modify it to `Kconfig_iso`, as I think that would be the ideal direction to to with the Kconfig files with named extensions.

File: subsys/bluetooth/Kconfig
Comment (Thalley) [inline subsys/bluetooth/Kconfig:180]:
```suggestion
rsource "Kconfig_iso"
```

**Review by Thalley — CHANGES_REQUESTED** [review]

**Comment by quic-egmc** [comment]
Reopening PR, we're *finally* getting around to addressing this.  Better late than never I guess?

Also adding @jonathonpenix who was willing to take this over the finish line, thanks!

**Comment by quic-egmc** [comment]
> @quic-egmc do you plan to return to this PR?

Yes but only about a year later.  :)

**Pull request closed without merging**
